### PR TITLE
Fix error handler redirecting

### LINKED ISSUE #269: Redirects can no longer be used in error blocks
First, thanks for all your work on Jester! It's a very nice framework. 

It seems that it's no longer possible to use a redirect within an error block with `0.5.0`. With Jester `0.4.3`, this snippet compiles and works as expected:

```
import jester

router root:
  get "/":
    resp "hello world"

  error Http404:
    redirect "404.html"

proc main() =
  var jester = initJester(root)
  jester.serve()

when isMainModule:
  main()
```
With `0.5.0`, I get the following compilation error:
```
/home/reesmichael1/programs/scorecord/example.nim(4, 8) template/generic instantiation of `router` from here
/home/reesmichael1/.nimble/pkgs/jester-0.5.0/jester.nim(1331, 39) template/generic instantiation of `async` from here
/home/reesmichael1/programs/scorecord/example.nim(9, 14) template/generic instantiation of `redirect` from here
/home/reesmichael1/.nimble/pkgs/jester-0.5.0/jester.nim(592, 11) Error: undeclared identifier: 'allRoutes'
```
If I change the `redirect` to a `resp`, the sample successfully compiles.

I haven't dug into it enough to be certain, but it seems likely that this is due to the changes in #265. Adding `halt = false` to the redirect call makes no difference.

### PR DESCRIPTION
**Pull request by iffy**
Fixes #269 (includes #267 because it relies on the testing changes included there)

`nimble test` doesn't work for me because of some of the other dependencies failing (as noted in #263), but I've tested the change and seen the added test pass.

### CODE REVIEW

**Comment by dom96** [comment]
Can you rebase so we can get a CI run?

**Comment by iffy** [comment]
Rebased; let me know if it looks good.

**Comment by dom96** [comment]
Thanks! Although seems the PR includes my commit, not sure what happened there

File: jester.nim [hunk 1]
```diff
@@ -451,18 +462,20 @@ proc initJester*(
   result.errorHandlers = @[]
 
 proc initJester*(
-  matcher: MatchProc,
+  pair: MatchPair,
```
Comment (dom96) [inline jester.nim:465]: Can we keep backwards compat by creating `initJester` overloads instead of editing these?

**Review by dom96 — APPROVED** [review]
Overall looks good. Just need slight change to be backwards compatible

File: jester.nim (hunk 1 above)
Comment (iffy) [reply jester.nim:465]: Oh, yeah, I think that would be fine. I guess there's no tests that test the existing behavior?

File: jester.nim [hunk 2]
```diff
@@ -477,6 +477,13 @@ proc initJester*(
   result.register(pair.matcher)
   result.register(pair.errorHandler)
 
+proc initJester*(
+  matcher: MatchProc,
```
Comment (dom96) [inline jester.nim:481]: Need the same for `MatchProcSync` as well

**Pull request merged**